### ISSUE TITLE
Skipped alcotest tests showing up as failures in Junit

### ISSUE THREAD

**Issue by 7h3kk1d** [Khady/ocaml-junit#9]
Example build for our repo: https://github.com/hazelgrove/hazel/actions/runs/12432886131/job/34713395956

Alcotest test output:
```
 [SKIP]        MenhirParser          0   Example failure.
 ```
 
 Junit.xml
 ```
 <testsuites>
    <testsuite package="HazelTests" id="0" name="HazelTests" timestamp="2024-12-20T14:40:16"
        hostname="localhost" tests="1" failures="0" errors="1" time="0">
        <properties></properties>
        <testcase name="Example failure" classname="HazelTests.MenhirParser" time="0">
            <error message="test crashed" type="exception raised">Alcotest_engine__Core.Skip</error>
        </testcase>
    </testsuite>
</testsuites>
```

I can work on a PR if that would be welcomed.

**Comment by Khady** [comment]
Thanks for the report! I'd be happy to accept a PR.

**Comment by Khady** [comment]
I've added a reproduction test in #10. It looks like alcotest doesn't expose the exception in its interface which makes it impossible to catch and properly handle :/

**Comment by Khady** [comment]
https://github.com/mirage/alcotest/issues/415 to have a better fix than what I did in #10.
